packages/pdf-viewer/hooks/useIsVisible.ts

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+
+
+const useIsVisible = (elementRef: React.MutableRefObject<HTMLElement>, rootRef: React.MutableRefObject<HTMLElement>) => {
+	const [isVisible, setIsVisible] = useState(false);
+	useEffect(() => {
+		let observer: IntersectionObserver = null;
+		if (elementRef.current) {
+			observer = new IntersectionObserver((entries, _observer) => {
+				let visable = false;
+				entries.forEach((entry) => {
+					if (entry.isIntersecting) {
+						visable = true;
+						setIsVisible(true);
+					}
+				});
+				if (!visable) {
+					setIsVisible(false);
+				}
+			}, {
+				root: rootRef.current,
+				rootMargin: '0px 0px 0px 0px',
+				threshold: 0,
+			});
+			observer.observe(elementRef.current);
+		}
+		return () => {
+			if (observer) {
+				observer.disconnect();
+			}
+		};
+	}, [elementRef]);
```

Comment:
no need to pass refs. When you access a ref inside a hook, you access elementRef.current. And that ".current" is always, as the name says, current, wherever it's accessed.